macantine/management/commands/generate_csat_diagnostics.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from django.core.management.base import BaseCommand
+
+from data.models import Diagnostic
+from macantine.utils import distribute_appro_values_between_satellites
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Generate diagnostics of CSAT by splitting diagnostics of central kitchens"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--year",
+            dest="year",
+            type=int,
+            required=True,
+            help="Year of the teledeclaration campaign to process",
+        )
+
+    def handle(self, *args, **options):
+        year = options["year"] | 2024
+        logger.info("Start task: generate diagnostics for satellite kitchens using central kitchen's diagnostics")
+
+        # Step 1: get all valid diagnostics for CC
+        diagnostics_cc = Diagnostic.objects.teledeclared_for_year(year=year).filter(
+            central_kitchen_diagnostic_mode__isnull=False
+        )
+
+        # Step 2: Fetch the satellites of the diag's central kitchen
+        for diag in diagnostics_cc:
+            satellites = diag.canteen.satellites
+            nbre_satellites = (
+                len(satellites) + 1
+                if diag.canteen.production_type == "Canteen.ProductionType.CENTRAL_SERVING"
```

Comment:
```suggestion
                if diag.canteen.production_type == Canteen.ProductionType.CENTRAL_SERVING
```
ca marchait avec les guillemets ?

Reply:
Nope, en effet